templates/fine-tune-llm_v2/cookbooks/long_context/README.ipynb

```diff
@@ -0,0 +1,115 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Fine-tuning on datasets with long context\n",
+    "**⏱️ Time to complete**: 90 minutes (assuming GCP A2 nodes can be acquired immidiately)\n",
+    "\n",
+    "This guide demonstrates how to prepare a dataset with long context lengths.\n",
+    "Make sure you have gone over the [basic fine-tuning guide](../../README.md) before going over this cookbook.\n",
+    "\n",
+    "In the following example, we filter a hugginface dataset and assemble it into the right format.\n",
+    "By estimating how many tokens an example from the dataset will result in, we can disgard examples that are too long.\n",
+    "You can use this as a template for creating your own datasets.\n",
+    "\n",
+    "> **_NOTE:_** To fine-tune with a context length of 8k tokens and the llama-3-8b.yaml config in this cookbook, we require GCP A2 nodes. For that, you need to instantiate your workspace in a region such as GCP's `us-central1`  due to the relatively high availability. See [GCP's availability info](https://cloud.google.com/compute/docs/gpus/gpu-regions-zones) for more information. If you only want to generate the dataset, you can do that anywhere."
```

Comment:
Again here, just mention the resource requirement. How they should get A100s (GCP or not) should be asked in a different thread and that's beyond the scope of this cookbook.

Reply:
Are you absolutely sure about this? If we remove this, users will have to find out themselves which cloud will give them a good chance of getting A100s. If we don't but it here at the start, they will create their workspace in AWS only to find out later that they need to create it again in GCP because you can't just move workspaces around.